examples/image_input/image_input.py

```diff
@@ -0,0 +1,49 @@
+import base64
+import os
+from typing import Union
+
+
+class ImageInput:
+    """
+    Represents a class with specific methods to help to prepare a list of image inputs for vision models api call.
+
+    Methods:
+    __init__(list_image_path: Union[str, None]):
+        Initializes a ImageInput instance with the provided attribute. Raises NotImplementedError if any attribute is None.
+    
+    _check_attributes():
+        Checks if the required attribute List_image_path is not None.
+        Raises NotImplementedError if the attribute is None.
+
+    encode() -> dict:
+        Encodes and returns a dictionary representation of the images encoded in base 64.
+
+    extension() -> dict:
+        Returns a dictionary of the extension of each image input (needed in the api call) : 
+
+        "type": "image_url", 
+           "image_url": {
+            "url": f"data:image/png;base64,{base64_images[0]}"}
+    """
+
+    def __init__(self, list_image_path: Union[list, None]):
```

Comment:
rename it images_paths or image_paths, with the "s" and the typing, it will be sufficient to know that it is a list